modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationConfiguration.java

```diff
@@ -0,0 +1,549 @@
+        addrRslvr = rslvr;
+    }
+
+    /**
+     * @return Local IP address.
+     */
+    public String localAddress() {
+        return locAddr;
+    }
+
+    /**
+     * @param locAddr New local IP address.
+     */
+    public void localAddress(String locAddr) {
+        this.locAddr = locAddr;
+    }
+
+    /**
+     * @return Local port which node uses.
+     */
+    public int localPort() {
+        return locPort;
+    }
+
+    /**
+     * @param locPort New local port which node uses.
+     */
+    public void localPort(int locPort) {
+        this.locPort = locPort;
+    }
+
+    /**
+     * @return Local port range.
+     */
+    public int localPortRange() {
+        return locPortRange;
+    }
+
+    /**
+     * @param locPortRange New local port range.
+     */
+    public void localPortRange(int locPortRange) {
+        this.locPortRange = locPortRange;
+    }
+
+    /**
+     * @return Local port which node uses to accept shared memory connections.
+     */
+    public int shmemPort() {
+        return shmemPort;
+    }
+
+    /**
+     * @param shmemPort New local port which node uses to accept shared memory connections.
+     */
+    public void shmemPort(int shmemPort) {
+        this.shmemPort = shmemPort;
+    }
+
+    /**
+     * @return Allocate direct buffer or heap buffer.
+     */
+    public boolean directBuffer() {
+        return directBuf;
+    }
+
+    /**
+     * @param directBuf New allocate direct buffer or heap buffer.
+     */
+    public void directBuffer(boolean directBuf) {
+        this.directBuf = directBuf;
+    }
+
+    /**
+     * @return Allocate direct buffer or heap buffer.
+     */
+    public boolean directSendBuffer() {
+        return directSndBuf;
+    }
+
+    /**
+     * @param directSndBuf New allocate direct buffer or heap buffer.
+     */
+    public void directSendBuffer(boolean directSndBuf) {
+        this.directSndBuf = directSndBuf;
+    }
+
+    /**
+     * @return Idle connection timeout.
+     */
+    public long idleConnectionTimeout() {
+        return idleConnTimeout;
+    }
+
+    /**
+     * @param idleConnTimeout New idle connection timeout.
+     */
+    public void idleConnectionTimeout(long idleConnTimeout) {
+        this.idleConnTimeout = idleConnTimeout;
+    }
+
+    /**
+     * @return Connect timeout.
+     */
+    public long connectionTimeout() {
+        return connTimeout;
+    }
+
+    /**
+     * @param connTimeout New connect timeout.
+     */
+    public void connectionTimeout(long connTimeout) {
+        this.connTimeout = connTimeout;
+    }
+
+    /**
+     * @return Maximum connect timeout.
+     */
+    public long maxConnectionTimeout() {
+        return maxConnTimeout;
+    }
+
+    /**
+     * @param maxConnTimeout New maximum connect timeout.
+     */
+    public void maxConnectionTimeout(long maxConnTimeout) {
+        this.maxConnTimeout = maxConnTimeout;
+    }
+
+    /**
+     * @return Reconnect attempts count.
```

Comment:
What does reconnects attempts means?
What will\won't happens if number of attempts exceeds configured value?